### PR TITLE
CosmosElement Continuation Tokens

### PR DESCRIPTION
**Pull request by bchong95**
# CosmosElement Continuation Tokens

## Description

Before we were using newtonsoft to serialize and deserialize continuation tokens. This leads to a bug when trying to work with extended types, since the newtonsoft serializer doesn't know how to handle them. 

I added ToCosmosElement and FromCosmosElement for orderby and composite continuation tokens to allow for this use case.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs [hunk 1]
```diff
@@ -12,14 +17,20 @@ namespace Microsoft.Azure.Cosmos.Query.Core.ContinuationTokens
...
+
+        [JsonProperty(TokenName)]
         public string Token
         {
             get;
             set;
         }
 
-        [JsonProperty("range")]
+        [JsonProperty(RangeName)]
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs]: What's its impact on compatibility?

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs]: It's the same string.

File: Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs [hunk 2]
```diff
@@ -31,5 +41,81 @@ public object ShallowCopy()
...
+                    },
+                });
+        }
+
+        public static TryCatch<CompositeContinuationToken> TryCreateFromCosmosElement(CosmosElement cosmosElement)
+        {
+            if (!(cosmosElement is CosmosObject cosmosObject))
+            {
+                return TryCatch<CompositeContinuationToken>.FromException(
+                    new MalformedContinuationTokenException($"{nameof(CompositeContinuationToken)} is not an object"));
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs]: Can we add the continuation token to the error message for easier troubleshooting?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationTokens/OrderByContinuationTokenTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    using Microsoft.Azure.Cosmos.Query.Core.ExecutionContext.OrderBy;
+    using Microsoft.Azure.Cosmos.Query.Core.Monads;
+    using Newtonsoft.Json;
+    using VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class OrderByContinuationTokenTests
+    {
+        [TestMethod]
+        public void TestRoundTripAsCosmosElement()
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationTokens/OrderByContinuationTokenTests.cs]: Should there be any tests for different cultures or is that already being tested?

File: Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ContinuationTokens/CompositeContinuationToken.cs]: Good idea

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationTokens/OrderByContinuationTokenTests.cs (hunk 3 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationTokens/OrderByContinuationTokenTests.cs]: We use Utf8Formatter, so there isn't any culture issues:

https://docs.microsoft.com/en-us/dotnet/api/system.buffers.text.utf8formatter?view=netcore-3.1

**Pull request merged**